### PR TITLE
add WithCustomInstaller method to Builder interface

### PR DESCRIPTION
**Pull request by alaypatel07**
There is a need in cluster etcd operator to check if it is safe
to install a revision on a node during install time.

Will allow to solve openshift/cluster-etcd-operator#278 with 
something like openshift/cluster-etcd-operator#281

### CODE REVIEW

File: pkg/operator/staticpod/controllers.go [hunk 1]
```diff
@@ -124,6 +129,15 @@ func (b *staticPodOperatorControllerBuilder) WithCerts(certDir string, certConfi
 
 func (b *staticPodOperatorControllerBuilder) WithInstaller(command []string) Builder {
 	b.installCommand = command
+	b.installerPodMutationFunc = func(pod *corev1.Pod, nodeName string, operatorSpec *operatorv1.StaticPodOperatorSpec, revision int32) error {
+		return nil
+	}
+	return b
+}
+
+func (b *staticPodOperatorControllerBuilder) WithMutatingInstallerPod(command []string, installerPodMutationFunc installer.InstallerPodMutationFunc) Builder {
```
Comment (sttts) [inline pkg/operator/staticpod/controllers.go]: what does this do? It needs docs.

Comment (sttts) [reply pkg/operator/staticpod/controllers.go]: MutatingInstallerPod sounds like the pod is mutating something. In reality we pass a func which mutates, right? `WithMutatedInstaller` or `WithCustomInstaller`?

Comment (alaypatel07) [reply pkg/operator/staticpod/controllers.go]: @sttts yes, WithCustomInstaller with do to explain makes more sense, will update

File: pkg/operator/staticpod/controllers.go [hunk 2]
```diff
@@ -25,6 +25,9 @@ import (
 	"github.com/openshift/library-go/pkg/operator/status"
 	"github.com/openshift/library-go/pkg/operator/unsupportedconfigoverridescontroller"
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
```
Comment (sttts) [inline pkg/operator/staticpod/controllers.go]: order

**Comment by hexfusion** [comment]
/title Bug 1817028: add WithCustomInstaller method to Builder interface

**Comment by hexfusion** [comment]
/retitle Bug 1817028: add WithCustomInstaller method to Builder interface

**Pull request merged**